### ISSUE TITLE
Faulty 'truncate' option in files_write()

### ISSUE THREAD

**Issue by lacabra** [ipfs/py-ipfs-api#112]
The following code fails:
```python
import io
import ipfsapi

api = ipfsapi.connect('localhost', 5001)

longtext = 'one two three'

res = api.files_write('/file.txt', io.BytesIO(str(longtext)), create=True)
assert api.files_read('/file.txt') == longtext

shorttext = 'this'

res = api.files_write('/file.txt', io.BytesIO(str(shorttext)), truncate=True)
assert api.files_read('/file.txt') == shorttext
```
The second assertion fails with `yes two three` ==  `yes`

This error is seems to be specific to the API implementation, as the command line equivalent works just fine:
```bash
$ echo 'one two three' | ipfs files write /file.txt --create
$ echo 'yes' | ipfs files write /file.txt --truncate
$ ipfs files read /file.txt
yes
```

**Comment by whereswaldon** [comment]
Yeah, that certainly seems like a bug in the implementation, thanks for pointing that out! Are you interested in trying to squash this? You've already provided something that essentially acts as a test case. All you've got to do is trace where that truncate option is getting lost in the code.

**Comment by rememberberry** [comment]
Not the OP, but the problem is on this line (missing quotes):
`opts = {"offset": offset, "create": create, truncate: truncate}`

**Comment by sujaypat** [comment]
@whereswaldon Can this issue be closed since #119 was merged?

**Comment by whereswaldon** [comment]
@sujaypat Absolutely.
